spec/paperclip/storage/ftp_spec.rb

```diff
@@ -22,218 +258,228 @@
+      it "yields the ftp servers, closes the connections afterwards" do
+        attachment.stub(:ftp_servers).and_return([first_server, second_server])
+        first_server.should_receive(:close_connection).ordered
+        second_server.should_receive(:close_connection).ordered
+        expect { |b| attachment.with_ftp_servers(&b) }.to yield_with_args([first_server, second_server])
+      end
     end
 
-    it "raises NoServerAvailable error if all servers are down" do
-      first_server.stub(:connected? => false)
-      second_server.stub(:connected? => false)
-      attachment.stub(:build_and_connect_server).and_return(first_server, second_server)
-      expect { attachment.ftp_servers }.to raise_error(Paperclip::Storage::Ftp::NoServerAvailable)
+    context "#ftp_servers" do
+      it "returns the configured ftp servers" do
+        first_server.stub(:connected? => true)
+        second_server.stub(:connected? => true)
+        attachment.stub(:build_and_connect_server).and_return(first_server, second_server)
+        attachment.ftp_servers.should == [first_server, second_server]
+      end
+
+      it "raises NoServerAvailable error if all servers are down" do
+        first_server.stub(:connected? => false)
+        second_server.stub(:connected? => false)
+        attachment.stub(:build_and_connect_server).and_return(first_server, second_server)
+        expect { attachment.ftp_servers }.to raise_error(Paperclip::Storage::Ftp::NoServerAvailable)
+      end
     end
-  end
 
-  context "#build_and_connect_server" do
-    it "returns a connected server instance based on the given options" do
-      Paperclip::Storage::Ftp::Server.stub(:new).and_call_original
-
-      expected_options = attachment.options[:ftp_servers].first.merge(
-        :connect_timeout       => attachment.options[:ftp_connect_timeout],
-        :ignore_connect_errors => attachment.options[:ftp_ignore_failing_connections]
-      )
-      expect(first_server).to receive(:establish_connection)
-      Paperclip::Storage::Ftp::Server.stub(:new).with(expected_options).and_return(first_server)
-      attachment.build_and_connect_server(attachment.options[:ftp_servers].first).should == first_server
+    context "#build_and_connect_server" do
+      it "returns a connected server instance based on the given options" do
+        Paperclip::Storage::Ftp::Server.stub(:new).and_call_original
+
+        servers = attachment.find_servers(attachment.options[:ftp_servers])
+        expected_options = servers.first.merge(
+          :connect_timeout       => attachment.options[:ftp_connect_timeout],
+          :ignore_connect_errors => attachment.options[:ftp_ignore_failing_connections]
+        )
+        expect(first_server).to receive(:establish_connection)
+        Paperclip::Storage::Ftp::Server.stub(:new).with(expected_options).and_return(first_server)
+        attachment.build_and_connect_server(servers.first).should == first_server
+      end
```

Comment:
Sorry, but duplicating almost all specs in here for this small new feature is not a great idea. 
I would reduce it to just testing (once) that a provided proc gets used by the `primary_ftp_server` and `ftp_servers` methods. 

Additionally a single new case should be added to `integration_spec.rb`.